File: src/Backend/FakeNews/FakeNews.Transfer/Articles/ArticleDto.cs
using FakeNews.Transfer.Categories;
using FakeNews.Transfer.Comment;
using System;
using System.Collections.Generic;

namespace FakeNews.Transfer.Articles
{
    public class ArticleDto
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string CreatedByUserId { get; set; }
        public string Content { get; set; }
        public bool ShownOnHomepage { get; set; }
        public DateTime CreatedDate { get; set; }
        public string CreatedByUserUserName { get; set; }
        public List<CategoryDto> Categories { get; set; }
        public List<CommentDto> Comments { get; set; }
    }
}

File: src/Frontend/fakenews/src/models/DTO/ArticleDto.ts
import { Moment } from 'moment';
import { CategoryDto } from './CategoryDto';
import { CommentDto } from './CommentDto';

export interface ArticleDto {
	id?: number;
	title: string;
	createdByUserId: string;
	content: string;
	shownOnHomepage: boolean;
	createdDate: Moment;
	validTo?: Moment;
	createdByUserUserName: string;
	categories: CategoryDto[];
	comments: CommentDto[];
}
